tests/token/erc1155/test_ERC1155MintableBurnable.py

```diff
@@ -0,0 +1,1468 @@
+            *uarr2cd(MINT_AMOUNTS),
+        ]
+    )
+
+
+@pytest.mark.asyncio
+async def test_mint_batch_to_zero_address(erc1155_factory):
+    erc1155, owner, _, _ = erc1155_factory
+
+    await assert_revert(
+        signer.send_transaction(
+            owner, erc1155.contract_address, 'mintBatch',
+            [ZERO_ADDRESS, *uarr2cd(TOKEN_IDS), *uarr2cd(MINT_AMOUNTS), DATA]),
+        "ERC1155: mint to the zero address")
+
+
+@pytest.mark.asyncio
+async def test_mint_batch_overflow(erc1155_factory):
+    erc1155, owner, account, _ = erc1155_factory
+
+    recipient = account.contract_address
+
+    # Bring recipient's balance to max possible
+    await signer.send_transaction(
+        owner, erc1155.contract_address, 'mintBatch',
+        [recipient, *uarr2cd(TOKEN_IDS), *uarr2cd(MAX_UINT_AMOUNTS), DATA])
+
+    # Issuing recipient any more on just 1 token_id
+    # should revert due to overflow
+    amounts = uint_array([0, 1, 0])
+    await assert_revert(
+        signer.send_transaction(
+            owner, erc1155.contract_address, 'mintBatch',
+            [recipient, *uarr2cd(TOKEN_IDS), *uarr2cd(amounts), DATA]),
+        "ERC1155: balance overflow")
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "amounts,token_ids,error",
+    [
+        (INVALID_AMOUNTS, TOKEN_IDS, "ERC1155: amount is not a valid Uint256"),
+        (MINT_AMOUNTS, INVALID_IDS, "ERC1155: id is not a valid Uint256")
+    ])
+async def test_mint_batch_invalid_uint(
+        erc1155_factory, amounts, token_ids, error):
+    erc1155, owner, account, _ = erc1155_factory
+
+    recipient = account.contract_address
+
+    await assert_revert(
+        signer.send_transaction(
+            owner, erc1155.contract_address, 'mintBatch',
+            [recipient, *uarr2cd(token_ids), *uarr2cd(amounts), DATA]),
+        error)
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "amounts,token_ids",
+    [
+        (MINT_AMOUNTS[:2], TOKEN_IDS),
+        (MINT_AMOUNTS, TOKEN_IDS[:2])
+    ])
+async def test_mint_batch_uneven_arrays(erc1155_factory, amounts, token_ids):
+    erc1155, owner, account, _ = erc1155_factory
+
+    recipient = account.contract_address
+
+    await assert_revert(
+        signer.send_transaction(
+            owner, erc1155.contract_address, 'mintBatch',
+            [recipient, *uarr2cd(token_ids), *uarr2cd(amounts), DATA]),
+        "ERC1155: ids and amounts length mismatch")
+
+
+@pytest.mark.asyncio
+async def test_mint_batch_to_receiver(erc1155_factory):
+    erc1155, owner, _, receiver = erc1155_factory
+
+    recipient = receiver.contract_address
+
+    await signer.send_transaction(
+        owner, erc1155.contract_address, 'mintBatch',
+        [
+            recipient, *uarr2cd(TOKEN_IDS),
+            *uarr2cd(MINT_AMOUNTS), DATA
+        ])
+
+    execution_info = await erc1155.balanceOfBatch(
+        [recipient]*3, TOKEN_IDS).invoke()
+    assert execution_info.result.balances == MINT_AMOUNTS
```

Comment:
again, i don't see the difference with `test_mint_batch` 🤔 . I understand the spirit is to test the acceptance of the token but I consider it implied in the above one, no?

Reply:
I wouldn't have said it is implied. In the first instance it is accepted by virtue of supporting the account interface, whereas in this case it is explicitly accepted by an `IERC1155Receiver` implementer. I could have in principle messed up either case and testing for just one wouldn't uncover that.